docs/en/styling-and-theming/supplemental.md

```diff
@@ -0,0 +1,642 @@
+
+## Structural widget styling
+
+Dojo leverages [CSS Modules](https://github.com/css-modules/css-modules) to provide all of the flexibility of CSS, but with the additional benefit of localized classes to help prevent inadvertent styling collisions across a large application. Dojo also makes use of [typed CSS modules](https://github.com/Quramy/typed-css-modules), allowing widgets to import their CSS modules similar to any other TypeScript module, and refer to specific CSS class names at design-time via IDE autocompletion.
+
+The CSS module file for a widget should have a `.m.css` extension, and by convention is usually named the same as the widget it is associated with.
+
+### Example
+
+Given the following CSS module file for a widget:
+
+> src/styles/MyWidget.m.css
+
+```css
+.myWidgetClass {
+	font-variant: small-caps;
+}
+
+.myWidgetExtraClass {
+	font-style: italic;
+}
+```
+
+This stylesheet can be used within a corresponding widget as follows:
+
+> src/widgets/MyWidget.ts
+
+```ts
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { v } from '@dojo/framework/widget-core/d';
+
+import * as css from '../styles/MyWidget.m.css';
+
+export default class MyWidget extends WidgetBase {
+	protected render() {
+		return v('div', { classes: [css.myWidgetClass, css.myWidgetExtraClass] }, ['Hello from a Dojo widget!']);
+	}
+}
+```
+
+Similarly, if using TSX widget syntax:
+
+> src/widgets/MyTsxWidget.tsx
+
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+
+import * as css from '../styles/MyWidget.m.css';
+
+export default class MyTsxWidget extends WidgetBase {
+	protected render() {
+		return <div classes={[css.myWidgetClass, css.myWidgetExtraClass]}>Hello from a Dojo TSX widget!</div>;
+	}
+}
+```
+
+When inspecting the CSS classes of these sample widgets in a built application, they will not contain `myWidgetClass` and `myWidgetExtraClass`, but rather obfuscated CSS class names similar to `MyWidget-m__myWidgetClass__33zN8` and `MyWidget-m__myWidgetExtraClass___g3St`.
+
+These obfuscated class names are localized to `MyWidget` elements, and are determined by Dojo's CSS modules build process. With this mechanism it is possible for other widgets in the same application to also use the `myWidgetClass` class name with different styling rules, and not encounter any conflicts between each set of styles.
+
+**Warning:** The obfuscated CSS class names should be considered unreliable and may change with a new build of an application, so developers should not explicitly reference them (for examples if attempting to target an element from elsewhere in an application).
+
+## Abstracting and extending stylesheets
+
+### CSS custom properties
+
+Dojo allows use of CSSnext features such as [custom properties and `var()`](http://cssnext.io/features/#custom-properties-var) to help abstract and centralize common styling properties within an application.
```

Comment:
We don't use cssnext anymore as it is deprecated, we use https://preset-env.cssdb.org/.